invenio/modules/deposit/cache.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2015 CERN.
+#
+# Invenio is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""Implementation of allowed deposit type caching."""
+
+from invenio.legacy.miscutil.data_cacher import DataCacher, DataCacherProxy
+
+
+class DepositTypeDataCacher(DataCacher):
+    def __init__(self):
+        def cache_filler():
+            from invenio.modules.access.control import acc_get_action_id
+            from invenio.modules.access.models import AccAuthorization, \
+                AccARGUMENT
+
+            action_id = acc_get_action_id('usedeposit')
+
+            return [auth[0] for auth in AccAuthorization.query.join(
```

Comment:
I think it might be causing an error. I get a following type error:
```
File "/home/stpk/envs/invenio/src/invenio/invenio/ext/login/legacy_user.py", line 84, in __init__
  acc = self._precache(data, force)
File "/home/stpk/envs/invenio/src/invenio/invenio/ext/login/legacy_user.py", line 203, in _precache
  get_authorized_deposition_types(user_info)
File "/home/stpk/envs/invenio/src/invenio/invenio/modules/deposit/cache.py", line 60, in get_authorized_deposition_types
  type=deposit_type_cacher.cache
File "/home/stpk/envs/invenio/src/invenio/invenio/modules/access/engine.py", line 57, in acc_authorize_action
  roles_list = acc_find_possible_roles(name_action, always_add_superadmin=True, batch_args=batch_args, **arguments)
File "/home/stpk/envs/invenio/src/invenio/invenio/modules/access/control.py", line 1451, in acc_find_possible_roles
  elem[k] = arguments[k][i]
TypeError: 'set' object does not support indexing
```